### ISSUE TITLE
"Access denied You are not authorized to access this page." after a new installation and trying to access the CMS through the internet / public network.

### ISSUE THREAD

**Issue by iconoclasta-digital** [backdrop/backdrop-issues#3555]
**Description of the bug**
A short description of what the problem is.
After installing for the first time, freshly and absolutely clean a new Backdrop CMS server, you try to login with your admin credentials via the **public network**, **a.k.a the internet**. You are able to land into the site, and enter the credentials, but, once you enter your credentials, you get the following error: 

_**"Access denied. You are not authorized to access this page."**_
****
If you log into the Backdrop CMS server via an Intranet or private network, you can access all the functionalities of Backdrop CMS server without any problems.

**Steps To Reproduce**
To reproduce the behavior:
1. Proceed to install a fresh, new computer with FreeBSD 12.0.
2. Proceed to configure MySQL 5.7.25 as your database server.
3. Proceed to configure Apache 2.4 as your web server.
4. Proceed to configure PHP 7.2 with the required modules that Backdrop needs for working as part of your web server. Make sure to edit the pertaining file associations into the httpd.conf file.
5. Create the backdrop_db with the appropriate character set and collation.
6. Create the backdrop_usr that will have full control of the backdrop_db.
7. Proceed to install, via download or via git, the latest version of the Backdrop CMS in your server. Make sure to extract the content into the Apache Web Server Data Root Directory and change the ownership of all the files so that Apache has full control of the files that you downloaded from the git repository or via download.
8. Proceed to reach / enter to your Backdrop CMS main site _through an INTRANET_, _Private network_ in order to setup the Backdrop CMS main site. _ANY attempt to try to do it through the INTERNET, Public network will destroy / render your steps 5 - 7 worthless_, in which case you will have to: 8.1) delete all of the files that were downloaded from the git repository, 8.2) drop the backdrop_db and recreate it again, 8.3) give full access to the backdrop_usr again to the backdrop_db. So, please, DON'T try to configure Backdrop via the Internet.
9. Make sure that your Web server complies with all the requirements stated by the Backdrop installer.
10. Create the administrator account with a secure password.
11. Once again, from the INTRANET, verify that you can login into the Backdrop server. Make sure that you can access all of the administrative sections and areas that the server offers.
12. Close your web browser and _now, this time, try to reach your Backdrop server through the INTERNET._
13. Proceed to enter the administrator's credentials that you created and verified in step 10-11.

**Actual behavior**
You will get an annoying message:
_**"Access denied. You are not authorized to access this page."**_
![screen shot 2019-02-26 at 0 04 28](https://user-images.githubusercontent.com/16010410/53346502-7a1be880-395a-11e9-8efe-40cbdde3313b.png)

**Expected behavior**
I was expecting to be able to enter the Backdrop CMS administration section / site to proceed to configure the site via Internet.
![screen shot 2019-02-26 at 0 11 45](https://user-images.githubusercontent.com/16010410/53346811-22ca4800-395b-11e9-81c3-f8145f3e7295.png)

**NOTICE**: This screenshot shows a working CMS Server from the INTRANET. Notice the Private Address on the navigation bar.

**Additional information**
Add any other information that could help, such as:
- Backdrop CMS version: 1.13.x-dev
- Web server and its version: Apache 2.4
- Operating System and its version: FreeBSD 12.0 p3
- Browser(s) and their versions: Chrome, Safari, Edge and any flavor of modern web browsers.

**Comment by klonos** [comment]
Thanks for taking the time to write this very detailed report @iconoclasta-digital 👍 ...sorry that this has been giving you trouble.

Having said that, I have never seen this problem before either installing locally or on a remote server. I will need to make time to try to reproduce this, although off of the top of my head, the only difference is that I have never installed on FreeBSD (which might be a factor, or not).

Curious to see if anyone else has seen this behavior before.

**Comment by iconoclasta-digital** [comment]
4 days and counting ....
Feels Bad, Man.

**Comment by laryn** [comment]
Do you have a $base_url or $cookie_domain set in `settings.php`? Just spitballing!

**Comment by findlabnet** [comment]
You can try to enable mod_rewrite, at least for avoid 404 on some points.

**Comment by hansfn** [comment]
I think laryn might been onto something - that the cookie for some reason has the wrong domain or something and hence you are not authorized even when the correct username/password is provided.

These types of problems is normally quite easy to track down if you have access to the server, but explaining all the things you should check and try in an issue/forum thread is hard and time consuming. 

If you create a test account for me, I could at least verify if the cookie is the problem.

Disclaimer: I don't use Backdrop, but I'm a long time D7 user who has used FreeBSD for many years.

**Comment by iconoclasta-digital** [comment]
> You can try to enable mod_rewrite, at least for avoid 404 on some points.

I already did. Since the installation process.

**Comment by iconoclasta-digital** [comment]
> I think laryn might been onto something - that the cookie for some reason has the wrong domain or something and hence you are not authorized even when the correct username/password is provided.
> 
> These types of problems is normally quite easy to track down if you have access to the server, but explaining all the things you should check and try in an issue/forum thread is hard and time consuming.
> 
> If you create a test account for me, I could at least verify if the cookie is the problem.
> 
> Disclaimer: I don't use Backdrop, but I'm a long time D7 user who has used FreeBSD for many years.

Please, send me an email and I can arrange to allow you access to that server. Actually, I am testing this setup in virtual machine, so at this stage what I am looking forward to is to make sure that the server is  capable of deployment.

Also, I forgot to mention that this server is BEHIND a Microsoft Forefront TMG Server 2010, which is a Firewall / Load Balancer / Reverse Proxy Server.

Never had issues with other platforms, but for some reason FreeBSD plays hard to get / hard to love.

We can actually arrange a communication process using voice / screen share to see what you are up to.

Any results about this problem will definitely benefit others.

**Comment by iconoclasta-digital** [comment]
> Do you have a $base_url or $cookie_domain set in `settings.php`? Just spitballing!

No, I don't. According to all the documentation that I have read, they never had mention anything about this, but it seems plausible and sensible to explicitly define them.

I'm going to test and come back to you guys and let you know what I found.

**Comment by iconoclasta-digital** [comment]
I've just read the documentation on Drupal 7 about setting up `$base_ur`l and / or `$cookie_domain` in `settings.php`

I understand that Backdrop is a fork of Drupal, but I wonder: If this step is so important, then why not mention it?

Thanks for following this thread and reading.

**Comment by iconoclasta-digital** [comment]
> I think laryn might been onto something - that the cookie for some reason has the wrong domain or something and hence you are not authorized even when the correct username/password is provided.
> 
> These types of problems is normally quite easy to track down if you have access to the server, but explaining all the things you should check and try in an issue/forum thread is hard and time consuming.
> 
> If you create a test account for me, I could at least verify if the cookie is the problem.
> 
> Disclaimer: I don't use Backdrop, but I'm a long time D7 user who has used FreeBSD for many years.

Please check your email. I sent you pertaining information.

**Comment by iconoclasta-digital** [comment]
Almost a month and counting. Do I really have to use Ubuntu or other Linux flavor just because Backdrop doesn't work in FreeBSD?

**Comment by hansfn** [comment]
I can guarantee you that Backdrop (or Drupal 7) works on FreeBSD. 

Anyway, I sent you an e-mail 14th of March asking if it was still relevant. Apparently it is, but you never replied. I'll have look at your server tonight if the credentials you sent still are valid.

**Comment by iconoclasta-digital** [comment]
Four months into this issue, and there is still no solution. No wonder why Microsoft and its products keeps being king in the IT world.

**Comment by iconoclasta-digital** [comment]
And this is a necro post, honestly. 14 months, that is, a year and 2 months and this issue is still open.
